src/components/Reown/NanoContractActions.js

```diff
@@ -61,40 +68,112 @@ const ActionItem = ({ action, isNft, title }) => {
+  const getActionIcon = (actionType) => {
+    const iconMap = {
+      [NanoContractActionType.DEPOSIT]: 'fa fa-arrow-up text-success',
+      [NanoContractActionType.WITHDRAWAL]: 'fa fa-arrow-down text-primary',
+      [NanoContractActionType.GRANT_AUTHORITY]: 'fa fa-arrow-up text-success',
+      [NanoContractActionType.INVOKE_AUTHORITY]: 'fa fa-arrow-down text-primary',
+    };
+
+    return iconMap[actionType] || 'fa fa-question-circle text-muted';
   };
 
+  // Check if this is an authority action
+  const isAuthorityAction = action.type === NanoContractActionType.GRANT_AUTHORITY
+    || action.type === NanoContractActionType.INVOKE_AUTHORITY;
+
+  // For authority actions, split the title to show authority type
+  const titleParts = isAuthorityAction && title.includes(':') ? title.split(':') : null;
+
   return (
     <div className="d-flex align-items-center p-3 border-bottom">
       <div className="mr-3">
-        {action.type === 'deposit' ? (
-          <i className="fa fa-arrow-up text-success"></i>
-        ) : (
-          <i className="fa fa-arrow-down text-primary"></i>
-        )}
+        <i className={getActionIcon(action.type)}></i>
       </div>
       <div className="flex-grow-1">
-        <div className="font-weight-bold">{title}</div>
-        {action.address && (
-          <div>
-            <small className="text-muted d-block">{t`To Address:`}</small>
-            <div className="text-monospace">
-              {action.address}
-              <button 
-                className="btn btn-link btn-sm p-0 ml-2" 
-                onClick={() => navigator.clipboard.writeText(action.address)}
-              >
-                <i className="fa fa-copy"></i>
-              </button>
-            </div>
+        {/* Title with authority type handling */}
+        {isAuthorityAction && titleParts ? (
+          <div className="d-flex justify-content-between align-items-center">
+            <div className="font-weight-bold">{titleParts[0].trim()}</div>
+            <div className="font-weight-bold text-muted">{titleParts[1].trim()}</div>
           </div>
+        ) : (
+          <div className="font-weight-bold">{title}</div>
         )}
+
+        {/* Grant Authority Address */}
+        {action.type === NanoContractActionType.GRANT_AUTHORITY
+          && (action.authorityAddress || action.address) && (
+            <div className="mt-2">
+              <small className="text-muted d-block">{t`Address to send a new Authority:`}</small>
+              <div className="text-monospace">
+                {action.authorityAddress || action.address}
+                <button
+                  className="btn btn-link btn-sm p-0 ml-2"
+                  onClick={() => navigator.clipboard.writeText(action.authorityAddress || action.address)}
+                >
+                  <i className="fa fa-copy"></i>
+                </button>
+              </div>
+            </div>
+          )}
```

Comment:
`authorityAddress` and `address` have different meanings. I think we should improve the UX for each parameter meaning.